src/coloring/acyclic_coloring.jl

```diff
@@ -41,30 +42,28 @@ function color_graph(g::LightGraphs.AbstractGraph, ::AcyclicColoring)
 
         color[v] = min_index(forbidden_colors, v)
 
-        # grow star for every edge connecting colored vertices v and w
         for w in outneighbors(g, v)
-            if color[w] != 0
-                grow_star!(set, v, w, g, first_neighbor, color)
+            if color[w]!=0
+                grow_star!(v, w, g, set,first_neighbor,color)
             end
         end
 
-        # merge the newly formed stars into existing trees if possible
         for w in outneighbors(g, v)
-            if color[w] != 0
+            if color[w]!=0
                 for x in outneighbors(g, w)
-                    if color[x] != 0 && x != v
-                        if color[x] == color[v]
-                            merge_trees!(set, v, w, x, g)
+                    if color[x]!=0 && x!=v
+                        if color[x]==color[v]
+                            merge_trees!(v,w,x,g,set)
                         end
                     end
                 end
             end
         end
     end
-
     return color
 end
 
+
 """
     prevent_cycle(v::Integer,
```

Comment:
Update the docstring for `preven_cycle`, add the "!" and check the function signature